### PR TITLE
Create sponsor table

### LINKED ISSUE #19: Create sponsor table for sponsor home page
**Description**
Utilize shadcn built in data table component to create the sponsor table (refer to sponsor home page figma wireframe for columns). Populate with test data.  Work with issue #21. https://ui.shadcn.com/docs/components/data-table

**Acceptance Criteria**

Steps:

- Research built in data table component from shadcn
- Modify data table to include required fields
- Populate table with test data
- Experiment with filtering and searching from shadcn

### PR DESCRIPTION
**Pull request by KimLinhVu**
## Developer: Kim-Linh Vu

Closes #19 

### Pull Request Summary

Created sponsor table component using data table component from shadcn. Added filtering by request name and pagination (next and prev buttons).

### Modifications

- `src/components/sponsor-table/columns.tsx`: added column information to use in data table component (request, requestFor, amount, status, requestDate, receipt)
- `src/components/sponsor-table/data-table.tsx`: created data table component from shadcn with filtering and pagination functionality
- `src/components/sponsor-table/sponsor-table.tsx`: created sponsor table component from data table component populated with test data

### Testing Considerations

Tested that table displays information correctly with mock data.

### Pull Request Checklist

- [x] Code is neat, readable, and works
- [x] Comments are appropriate
- [x] The commit messages follows our [guidelines](https://h4i.notion.site/Conventional-Commits-593452ad1179489399ad3bd696ef772a)
- [x] The developer name is specified
- [x] The summary is completed
- [x] Assign reviewers

### Screenshots/Screencast

Screenshot 2024-02-09 at 5 48 43 PM

### CODE REVIEW

**Review by s-iloo — DISMISSED** [review]
Looks great Kim-Linh! Awesome job.

File: src/components/sponsor-table/columns.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+"use client";
+
+import { ColumnDef } from "@tanstack/react-table";
+
+export type Payment = {
```
Comment (s-iloo) [inline src/components/sponsor-table/columns.tsx]: Super minor change but once the Reimbursement schema is merged into develop, could you refer to the schema instead? Thanks!

File: src/components/sponsor-table/columns.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
+"use client";
+
+import { ColumnDef } from "@tanstack/react-table";
+import { DownloadIcon } from "@radix-ui/react-icons";
+import { Button } from "@/components/ui/button";
+
+export type Reimbursement = {
```
Comment (s-iloo) [inline src/components/sponsor-table/columns.tsx]: You should be able to use the interface specified in the Reimbursement schema now.

Comment (KimLinhVu) [reply src/components/sponsor-table/columns.tsx]: done

**Review by ryanhu021 — APPROVED** [review]

**Pull request merged**